CMakeLists.txt

```diff
@@ -35,1037 +28,139 @@ link_libraries(Threads::Threads)
-    Inferences/TermAlgebraReasoning.cpp
-    Inferences/URResolution.cpp
-    Inferences/DefinitionIntroduction.cpp
-    Inferences/BackwardDemodulation.hpp
-    Inferences/BackwardSubsumptionAndResolution.hpp
-    Inferences/BinaryResolution.hpp
-    Inferences/CodeTreeForwardSubsumptionAndResolution.hpp
-    Inferences/Condensation.hpp
-    Inferences/DemodulationHelper.hpp
-    Inferences/DistinctEqualitySimplifier.hpp
-    Inferences/EqualityFactoring.hpp
-    Inferences/EqualityResolution.hpp
-    Inferences/ExtensionalityResolution.hpp
-    Inferences/Factoring.hpp
-    Inferences/FastCondensation.hpp
-    Inferences/FOOLParamodulation.hpp
-    Inferences/ForwardDemodulation.hpp
-    Inferences/ForwardLiteralRewriting.hpp
-    Inferences/ForwardSubsumptionAndResolution.hpp
-    Inferences/FunctionDefinitionRewriting.hpp
-    Inferences/GlobalSubsumption.hpp
-    Inferences/InnerRewriting.hpp
-    Inferences/EquationalTautologyRemoval.hpp
-    Inferences/InductionHelper.hpp
-    Inferences/InferenceEngine.hpp
-    Inferences/Instantiation.hpp
-    Inferences/InterpretedEvaluation.hpp
-    Inferences/InvalidAnswerLiteralRemovals.hpp
-    Inferences/PushUnaryMinus.hpp
-    Inferences/Cancellation.hpp
-    Inferences/ArithmeticSubtermGeneralization.hpp
-    Kernel/NumTraits.cpp
-    Inferences/GaussianVariableElimination.hpp
-    Kernel/Rebalancing.hpp
-    Kernel/Rebalancing/Inverters.hpp
-    Inferences/SubsumptionDemodulationHelper.cpp
-    Inferences/Superposition.hpp
-    Inferences/TautologyDeletionISE.hpp
-    Inferences/TermAlgebraReasoning.hpp
-    Inferences/URResolution.hpp
-    Inferences/DefinitionIntroduction.hpp
-    Inferences/TheoryInstAndSimp.hpp
-    Inferences/TheoryInstAndSimp.cpp  # this is theory instantiation
-    Inferences/ArithmeticSubtermGeneralization.hpp
-    Inferences/ArithmeticSubtermGeneralization.cpp
-    Inferences/PolynomialEvaluation.hpp
-    Inferences/PolynomialEvaluation.cpp
-    Inferences/Cancellation.hpp
-    Inferences/Cancellation.cpp
-    )
-source_group(inference_source_files FILES ${VAMPIRE_INFERENCE_SOURCES})
-
-set(VAMPIRE_SAT_SOURCES
-    SAT/BufferedSolver.cpp
-    SAT/FallbackSolverWrapper.cpp
-    SAT/MinimizingSolver.cpp
-    SAT/SAT2FO.cpp
-    SAT/SATClause.cpp
-    SAT/SATInference.cpp
-    SAT/SATLiteral.cpp
-    SAT/CadicalInterfacing.cpp
-    SAT/Z3Interfacing.cpp
+# add top level directory to the search path of compiler
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-    SAT/BufferedSolver.hpp
-    SAT/FallbackSolverWrapper.hpp
-    SAT/MinimizingSolver.hpp
-    SAT/SAT2FO.hpp
-    SAT/SATClause.hpp
-    SAT/SATInference.hpp
-    SAT/SATLiteral.hpp
-    SAT/SATSolver.hpp
-    SAT/CadicalInterfacing.hpp
-    SAT/Z3Interfacing.hpp
-    )
-source_group(sat_source_files FILES ${VAMPIRE_SAT_SOURCES})
+# enable for (slightly) more deallocation
+add_compile_definitions(CHECK_LEAKS=0)
+# enable for time profiling
+add_compile_definitions(VTIME_PROFILING=0)
```

Comment:
What do you think about guarding these with 
```
if(DEFINED ENV{CHECK_LEAKS})
  add_compile_definitions(CHECK_LEAKS="$ENV{CHECK_LEAKS}")
else()
    add_compile_definitions(CHECK_LEAKS=0)
endif()
```
(and same for `VTIME_PROFILING`)

Reply:
Good idea - I'll make them an option so we don't have to go around editing `CMakeLists` to activate them. Or did you specifically want an environment variable?

`CHECK_LEAKS` is kind of silly now, anyhow - I'll put it on my list to investigate better memory leak tracking.